packages/ui/src/containers/CreateAccount/index.tsx

```diff
@@ -26,119 +30,46 @@ type FieldState = {
   confirmPassword: string;
 };
 
-type ErrorState = {
-  [key in keyof FieldState]?: ErrorType;
-};
-
-type FieldValidations = {
-  [key in keyof FieldState]?: (state: FieldState) => ErrorType | undefined;
-};
-
 type Props = {
   className?: string;
 };
 
-const defaultState = {
+const defaultState: FieldState = {
   username: '',
   password: '',
   confirmPassword: '',
 };
 
-const usernameRegx = /^[A-Z_a-z-][\w-]*$/;
-
 const CreateAccount = ({ className }: Props) => {
   const { t, i18n } = useTranslation(undefined, { keyPrefix: 'main_flow' });
-  const [fieldState, setFieldState] = useState<FieldState>(defaultState);
-  const [fieldErrors, setFieldErrors] = useState<ErrorState>({});
   const { termsValidation } = useTerms();
-
   const { setToast } = useContext(PageContext);
-
   const { error, result, run: asyncRegister } = useApi(register);
-
-  const validations = useMemo<FieldValidations>(
-    () => ({
-      username: ({ username }) => {
-        if (!username) {
-          return { code: 'required', data: { field: t('input.username') } };
-        }
-
-        if (/\d/.test(username.slice(0, 1))) {
-          return 'username_should_not_start_with_number';
-        }
-
-        if (!usernameRegx.test(username)) {
-          return 'username_valid_charset';
-        }
-      },
-      password: ({ password }) => {
-        if (!password) {
-          return { code: 'required', data: { field: t('input.password') } };
-        }
-
-        if (password.length < 6) {
-          return { code: 'password_min_length', data: { min: 6 } };
-        }
-      },
-      confirmPassword: ({ password, confirmPassword }) => {
-        if (password !== confirmPassword) {
-          return { code: 'passwords_do_not_match' };
-        }
-      },
-    }),
-    [t]
-  );
+  const {
+    fieldValue,
+    setFieldValue,
+    register: fieldRegister,
+    formValidation,
+  } = useForm(defaultState);
 
   const onSubmitHandler = useCallback(() => {
-    // Validates
-    const usernameError = validations.username?.(fieldState);
-    const passwordError = validations.password?.(fieldState);
-
-    if (usernameError || passwordError) {
-      setFieldErrors((previous) => ({
-        ...previous,
-        username: usernameError,
-        password: passwordError,
-      }));
-
-      return;
-    }
-
-    const confirmPasswordError = validations.confirmPassword?.(fieldState);
-
-    if (confirmPasswordError) {
-      setFieldErrors((previous) => ({ ...previous, confirmPassword: confirmPasswordError }));
-
+    if (!formValidation()) {
```

Comment:
```suggestion
    if (!validateForm()) {
```
use verb for function names?